modules/node-shared/src/main/scala/org/tessellation/node/shared/domain/seedlist/HistoricalSnapshotOrdinalEntries.scala

```diff
@@ -0,0 +1,93 @@
+package org.tessellation.node.shared.domain.seedlist
+
+import cats.syntax.all._
+
+import org.tessellation.schema.SnapshotOrdinal
+
+import derevo.cats.eqv
+import derevo.derive
+import eu.timepit.refined.auto._
+import eu.timepit.refined.numeric.NonNegative
+import eu.timepit.refined.refineV
+import fs2.data.csv.{CellDecoder, DecoderError}
+
+@derive(eqv)
+case class HistoricalSnapshotOrdinalEntries(values: List[HistoricalSnapshotOrdinalEntry])
+
+object HistoricalSnapshotOrdinalEntries {
+
+  implicit val historicalSnapshotOrdinalEntriesCellDecoder: CellDecoder[Option[HistoricalSnapshotOrdinalEntries]] =
+    CellDecoder.stringDecoder.emap { s =>
+      val timeline = mkSnapshotOrdinalTimeline(s)
+
+      if (timeline.isDefined)
+        timeline.asRight
+      else
+        new DecoderError("Unable to parse the ordinal timeline").asLeft
+    }
+
+  private val valueDelimiter = "-"
+  private val entryDelimiter = ";"
+
+  def deserialize(value: String): Either[String, Array[Either[String, (Option[Long], Option[Long])]]] =
+    if (value.contains("--"))
+      "Invalid ranges string".asLeft
+    else {
+      val ranges = value
+        .split(";")
+        .map(_.split("-"))
+        .map {
+          case Array(start, end, _*) if start.toLongOption.isDefined => (start.toLongOption, end.toLongOption).asRight
+          case Array(start, _*) if start.toLongOption.isDefined      => (start.toLongOption, none).asRight
+          case _                                                     => "Should never happen".asLeft
```

Comment:
"Should never happen" -> "Unpexpected state"

Reply:
The code has been re-worked to not need this.